### ISSUE TITLE
Support for new Steam Chat feature

### ISSUE THREAD

**Issue by Duckbuster** [Philipp15b/go-steam#94]
Valve released a new chat a few months ago:
https://steamcommunity.com/updates/chatupdate

It's new and all web-based now.

It has new group chat settings, looks more like Discord/slack now.
For example:
-Group chats can now have multiple text channels and voice channels.
-Steam media upload (pictures) and embedding of content.

Any idea if you will add support for this new chat? Or if it's even possible?

Because your steam implementation gets used by matterbridge and that is broken for now (https://github.com/42wim/matterbridge/issues/457#issuecomment-400850379)

**Comment by Philipp15b** [comment]
I didn't know this library was used in matterbridge, that's cool!
Unfortunately, reverse engineering of the chat API seems to be an ongoing problem: https://github.com/SteamRE/SteamKit/issues/561.
It may be possible to just use the web API instead of the protobuf format, but that also seems like a lot of work.
I'll try to look into this in the next few days.

**Comment by Philipp15b** [comment]
I don't think I have enough time for this. Maybe once the issue is fixed in SteamKit it will be easier.

**Comment by poVoq** [comment]
https://github.com/SteamRE/SteamKit/releases

Looks like there is movement on the Steamkit side.

Together with the new Steam chat app it would be really nice if Matterbridge Steam support would work again.

**Comment by LucesEnBabel** [comment]
Any news about this? [Apparently it's already possible to interact with the new Steam Chat system](https://github.com/SteamRE/SteamKit/issues/561#issuecomment-549092906).

**Comment by RarogCmex** [comment]
Any news?

**Comment by Philipp15b** [comment]
Not really. If I am not mistaken, https://github.com/SteamRE/SteamKit/issues/561 is still open and there is no implementation of the new chat in SteamKit or anywhere else to my knowledge. I don't have the bandwidth to reverse engineer and implement this myself.

**Comment by cooljeanius** [comment]
The SuperTux and Wesnoth communities could both have uses for this issue being fixed

**Comment by dylan-tock** [comment]
In case you weren't aware, here are a couple of recent posts from [SteamRE/SteamKit#561](https://github.com/SteamRE/SteamKit/issues/561):
* [Matrix <-> Steam bridge](https://github.com/icewind1991/mx-puppet-steam)
* [Information about the SteamChatRoomClient protocol/objects/methods](https://github.com/DoctorMcKay/node-steam-user/wiki/SteamChatRoomClient) that may be what's used by mx-puppet-steam.
